File: java/src/main/java/com/hyperswitchai/examples/AddStrategy.java
package com.hyperswitchai.examples;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.HashMap;
import java.util.Map;

public class AddStrategy {
    private static final String BASE_URL = "https://api.hyperswitchai.com";
    private static final ObjectMapper mapper = new ObjectMapper();
    private static final HttpClient client = HttpClient.newHttpClient();

    public static void main(String[] args) {
        try {
            String token = TokenManager.getToken();

            // Add a new strategy
            Map<String, String> strategyPayload = new HashMap<>();
            strategyPayload.put("code", "claude-proxy-1");
            strategyPayload.put("type", "proxy");
            strategyPayload.put("model", "claude-3-5-sonnet-20240620");
            strategyPayload.put("keyId", "claude-key-1");
            strategyPayload.put("provider", "anthropic");

            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + "/admin/strategies/add-strategy"))
                .header("Content-Type", "application/json")
                .header("Authorization", "Bearer " + token)
                .POST(HttpRequest.BodyPublishers.ofString(mapper.writeValueAsString(strategyPayload)))
                .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            if (response.statusCode() == 409) {
                try {
                    Map<String, Object> errorResponse = mapper.readValue(response.body(), Map.class);
                    Map<String, String> error = (Map<String, String>) errorResponse.get("error");
                    if (error != null && "duplicate_strategy_code".equals(error.get("code"))) {
                        System.out.println("\n\u001B[33m⚠️  Strategy already exists with this code\u001B[0m");
                        return;
                    }
                } catch (Exception e) {
                    // If JSON parsing fails, continue with error handling
                }
            }

            if (response.statusCode() != 200) {
                System.out.println("\u001B[31mServer response: " + response.statusCode() + " " + response.body() + "\u001B[0m");
                throw new RuntimeException("Failed to add strategy: " + response.body());
            }

            System.out.println("\u001B[32m✅ Strategy added successfully\u001B[0m");

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
} 

File: csharp/HyperSwitchAI/src/AddStrategy.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using DotNetEnv;

namespace HyperSwitchAI
{
    public class AddStrategyRequest
    {
        public string? code { get; set; }
        public string? type { get; set; }
        public string? model { get; set; }
        public string? keyId { get; set; }
        public string? provider { get; set; }
    }

    public class AddStrategy
    {
        private const string BASE_URL = "https://api.hyperswitchai.com";

        public static async Task RunAsync()
        {
            try
            {
                Env.Load();
                var token = await CacheToken.GetToken();

                // Add a new strategy
                using (var client = new HttpClient())
                {
                    var addStrategyRequest = new AddStrategyRequest
                    {
                        code = "claude-proxy-1",
                        type = "proxy",
                        model = "claude-3-5-sonnet-20240620",
                        keyId = "claude-key-1",
                        provider = "anthropic"
                    };

                    var json = JsonSerializer.Serialize(addStrategyRequest);
                    var content = new StringContent(json, Encoding.UTF8, "application/json");

                    client.DefaultRequestHeaders.Add("Authorization", $"Bearer {token}");
                    var response = await client.PostAsync($"{BASE_URL}/admin/strategies/add-strategy", content);

                    if (!response.IsSuccessStatusCode)
                    {
                        var responseData = await response.Content.ReadAsStringAsync();

                        if (response.StatusCode == System.Net.HttpStatusCode.Conflict)
                        {
                            try
                            {
                                var error = JsonSerializer.Deserialize<ErrorResponse>(responseData);
                                if (error?.error?.code == "duplicate_strategy_code")
                                {
                                    Console.WriteLine("\n\u001b[33m⚠️  Strategy already exists with this code\u001b[0m");
                                    return;
                                }
                            }
                            catch
                            {
                                // If JSON parsing fails, continue with the text response
                            }
                        }

                        Console.WriteLine($"\n\u001b[31mServer response: {response.StatusCode} {responseData}\u001b[0m");
                        throw new Exception($"Failed to add strategy: {response.StatusCode}");
                    }

                    Console.WriteLine("\n\u001b[32m✅ Strategy added successfully\u001b[0m");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }
    }
}
